lib/Models/MagdaReference.ts

```diff
@@ -168,33 +169,37 @@ export default class MagdaReference extends AccessControlMixin(
     const override = toJS(this.override);
     const distributionFormats = this.preparedDistributionFormats;
 
-    const target = MagdaReference.createMemberFromRecord(
-      this.terria,
-      this,
-      distributionFormats,
-      magdaUri,
-      this.uniqueId,
-      existingRecord,
-      override,
-      previousTarget
-    );
-    if (target !== undefined) {
-      return Promise.resolve(target);
-    }
+    // `runLater` is needed due to no actions in `AsyncLoader` computed promise (See AsyncLoader.ts)
```

Comment:
is this to turn the provided function into an async function as it will eventually be passed in as a loadCallback function for AsyncLoader?

Reply:
Yeah 😢

Reply:
oh wait

Reply:
`return Promise.resolve(target);` can be `return target;